### DISCUSSION TITLE
Slats tilting with venetian blinds

### DISCUSSION THREAD

**Discussion by Nagskull** [zwave-js/zwave-js-ui#2048] [Q&A]
Hello,

I have 9 Qubino modules (ZMNHCD1) to control venetian blinds. They are really difficult to use because their position and their inclination cannot be set simultaneously: you must set their position, wait until they have reached it, then set their inclination. The problem is that, when a position is set, zwavejs2mqtt immediately reports that it has been reached, i.e., it equates the currentValue with the targetValue without waiting a report from the module.

I am not the first one to experience this problem: https://community.home-assistant.io/t/z-wave-js-cover-state-feedback/298598

This seems to be an intented behavior: https://github.com/home-assistant/core/issues/47950#issuecomment-876035703

OptimisticValueUpdate make sense in the case of lights. For shutters or blinds with significant moving times, e.g., almost 60 seconds in the case of mine, it seems very wrong. Is there a workaround? Could it be possible to disable that option on a node by node basis (i.e., for shutter but not light modules)?

Feel free to tell me if this is not the right place to discuss this, or more details are needed. This is my first post on github
 and I did my best! Thanks,

**Comment by robertsLando** [comment]
@Nagskull I'm not sure here but if this happens it means your device is reporting the current value immediatly instead of reporting it when it reaches the position so nothing that could be handled by me :( BTW maybe @AlCalzone can say more about this

**Reply by AlCalzone** [reply]
@Nagskull I agree it makes sense for slow devices to not optimistically update.
I've asked the HA devs to chime in, because it may have some more implications.

Can you share a node dump please?

**Reply by Nagskull** [reply]
Thanks for your answer, and all your hard work on this project!

I don’t think the problem is related neither to HA nor to the devices. I am personally using Jeedom: I have no problem with the old zwave integration that relies on openzwave 1.4, because the devices only send reports when they have reached the desired position. With zwavejs2mqtt, I have noticed that whenever I set a new targetValue, the currentValue is instantly updated, while the blinds have not even started to move. I don’t know what could cause this but optimisticvalueUpdate. I’ll see for a node dump this evening. Let me know if you need any other information.

Of course, the best solution would be for Quibino devices to accept position and inclination commands simultaneously and compute the relevant waiting time themselves. This would make life so much easier for us owners of venetian blinds! I have opened a ticket with the Qubino support, but I don’t know if that is possible, and if they would accept to provide a new firmware.

**Reply by Nagskull** [reply]
Here is the [json](https://github.com/zwave-js/zwavejs2mqtt/files/7706835/node_4.txt) from the node, if that is indeed what you need,  and a copy from the log when I set the vertical position of the blinds.

> 2021-12-13 20:32:26.959 INFO MQTT: Message received on zwave/salon/jalousie_2/38/1/targetValue/set: '78'
> 2021-12-13 20:32:26.963 INFO ZWAVE: Writing 78 to 4-38-1-targetValue
> 2021-12-13 20:32:27.218 DEBUG MQTT: Publishing to zwave/salon/jalousie_2/38/1/currentValue: { time: 1639427547218, value: 78 } with options { qos: 1, retain: true }
> 2021-12-13 20:32:27.219 INFO ZWAVE: Node 4: value updated: 38-1-currentValue 99 => 78
> 2021-12-13 20:32:27.223 DEBUG MQTT: Publishing to zwave/salon/jalousie_2/38/1/targetValue: { time: 1639427547223, value: 78 } with options { qos: 1, retain: true }
> 2021-12-13 20:32:27.224 INFO ZWAVE: Node 4: value updated: 38-1-targetValue 85 => 78
> 2021-12-13 20:32:32.652 DEBUG MQTT: Publishing to zwave/salon/jalousie_2/38/1/currentValue: { time: 1639427552652, value: 93 } with options { qos: 1, retain: true }
> 2021-12-13 20:32:32.654 INFO ZWAVE: Node 4: value updated: 38-1-currentValue 78 => 93

The log shows another problem, incidentally: the targetValue was 78, but the currentValue that is finally transmitted in 93. I need to manually update the device to get the right value. I am not sure what is happening here, but with openzwave 1.4., the correct position is reported at the end of the move.

Thanks for your help,

**Reply by AlCalzone** [reply]
An idea could be to better distinguish between fast and slow transitions and only do optimistic updates for fast ones (which is where UIs might want to give immediate feedback).

Fast would be:
- Lights without a specified transition time
- Lights with a short transition time (< 5s for example)
- Binary switches, etc.

Slow would be:
- Motor control (which your device identifies as)
- Window Covering

**Reply by Nagskull** [reply]
That sounds promising!

**Reply by AlCalzone** [reply]
Follow https://github.com/zwave-js/node-zwave-js/issues/3892 for updates
